numpy/lib/function_base.py

```diff
@@ -1801,6 +1801,10 @@ def place(arr, mask, vals):
            [44, 55, 44]])
 
     """
+    if not isinstance(arr, np.ndarray):
```

Comment:
@shoyer : Sorry, I didn't answer the remainder of your question; I only had seen the "It might be worth" part on my phone when I was reading it.  ```place``` doesn't actually do any duck-typing if you pass in a non-```ndarray``` to it.  Rather, it does nothing, which is why I added that ```isinstance``` check to the code.  However, #7003 will embed that check into the C code, which is preferable.